### PR TITLE
ART-10508: add Jira and PR creation logic with submitRequest button

### PR DESCRIPTION
**Pull request by DennisPeriquet**
This calls the new /git_jira_api endpoint introduced in [this PR](https://github.com/openshift-eng/art-dashboard-server/pull/150) in the art-ui-server backend to create the PR and Jira for image add requests.

That PR must merge first.

### CODE REVIEW

**Comment by DennisPeriquet** [comment]
/hold

This cannot merge until the `/git_jira_api` is merged, up and functional.

File: components/self-service/new-content-form.tsx
Comment (ashwindasr) [inline components/self-service/new-content-form.tsx:121]: Can we dynamically get the highest branch on `ocp-build-data`, like `openshift-4.19` at this point of time?

File: components/self-service/new-content-done.tsx
Comment (ashwindasr) [inline components/self-service/new-content-done.tsx:238]:
```suggestion
          Submit Request
```
nit: add a space between the words

File: components/self-service/new-content-done.tsx
Comment (ashwindasr) [inline components/self-service/new-content-done.tsx:227]:
```suggestion
        <li>If the above looks good, click Submit Request and a Jira and PR will be created</li>
```
same nit

**Review by ashwindasr — CHANGES_REQUESTED** [review]
Amazing work @DennisPeriquet! Thanks for working on this!

Minor nits in this PR, but the major issue is that we should restrict the number of times the JIRA and PR creation requests are sent it (i.e. to only once, as mentioned in [Slack](https://redhat-internal.slack.com/archives/C07T08DJWRE/p1731077012782389))

File: components/self-service/new-content-form.tsx
Comment (DennisPeriquet) [reply components/self-service/new-content-form.tsx]: this is now implemented

File: components/self-service/new-content-done.tsx
Comment (DennisPeriquet) [reply components/self-service/new-content-done.tsx]: fixed

**Comment by DennisPeriquet** [comment]
re: the case where the api call to create the PR and Jira fails, ... if this happens, we now keep the "Submit Request" button up so that user can re-click the button and retry without having to re-enter all of the values they set earlier.

**Review by ashwindasr — APPROVED** [review]
Server [PR](https://github.com/openshift-eng/art-dashboard-server/pull/150) has merged

**Pull request merged**
